src/corehost/src/main.cpp

```diff
@@ -179,46 +203,26 @@ int main(const int argc, const pal::char_t* argv[])
     trace::info(_X("App argc: %d"), args.app_argc);
     trace::info(_X("App argv: %s"), argstr.c_str());
 
-    auto app_base = get_directory(args.managed_application);
-    auto app_name = get_filename(args.managed_application);
-
-    // App-local coreclr wins
-    {
-        pal::string_t candidate;
-        candidate.assign(app_base);
-        append_path(candidate, LIBCORECLR_NAME);
-
-        if (pal::file_exists(candidate))
-        {
-            args.clr_path.assign(app_base);
-        }
-    }
-
-    if (args.clr_path.empty())
+    pal::string_t clr_path;
+    if (!resolve_clr_path(args, &clr_path))
     {
-        trace::error(_X("failed to locate CLR files, set the DOTNET_HOME environment variable"));
+        trace::error(_X("could not resolve coreclr path"));
         return 1;
     }
 
-    if (!pal::realpath(args.clr_path))
-    {
-        trace::error(_X("failed to locate CLR files at %s"), args.clr_path.c_str());
-        return 1;
-    }
-
-    trace::info(_X("using CLR files from: %s"), args.clr_path.c_str());
-    trace::info(_X("preparing to launch: %s"), app_name.c_str());
-    trace::info(_X("using app base: %s"), app_base.c_str());
-
     // Check for and load deps file
     pal::string_t tpafile_path;
-    get_tpafile_path(app_base, app_name, tpafile_path);
-    trace::info(_X("checking for .deps File at: %s"), tpafile_path.c_str());
-    tpafile tpa;
+    auto app_name = get_filename(args.managed_application);
+    get_tpafile_path(args.app_dir, app_name, &tpafile_path);
```

Comment:
I think "get_depsfile_path" should be an private function of tpafile type. Initialization of tpafile should take args and, also, internally construct servicing_index_t instance it is not being used by anyone but tpafile implementation.